ocs_ci/ocs/resources/osd_resize.py

```diff
@@ -0,0 +1,306 @@
+        )
+
+    ceph_cluster = CephCluster()
+    ceph_capacity = ceph_cluster.get_ceph_capacity()
+    expected_storage_size_in_gb = convert_device_size(expected_storage_size, "GB")
+    logger.info(
+        f"Check that the Ceph capacity {ceph_capacity} is equal "
+        f"to the expected storage size {expected_storage_size_in_gb}"
+    )
+    if not int(ceph_capacity) == expected_storage_size_in_gb:
+        raise StorageSizeNotReflectedException(
+            f"The Ceph capacity {ceph_capacity} is not equal to the "
+            f"expected storage size {expected_storage_size_in_gb}"
+        )
+
+
+def check_ceph_state_post_resize_osd():
+    """
+    Check the Ceph state post resize osd.
+    The function checks the Ceph device classes and osd tree.
+
+    Raises:
+        CephHealthException: In case the Ceph device classes and osd tree checks
+            didn't finish successfully
+
+    """
+    logger.info("Check the Ceph device classes and osd tree")
+    device_class = get_device_class()
+    ct_pod = get_ceph_tools_pod()
+    try:
+        verify_storage_device_class(device_class)
+        verify_device_class_in_osd_tree(ct_pod, device_class)
+    except AssertionError as ex:
+        raise CephHealthException(ex)
+    if not check_ceph_osd_tree():
+        raise CephHealthException("The ceph osd tree checks didn't finish successfully")
+
+
+def base_ceph_verification_steps_post_resize_osd(
+    old_osd_pods, old_osd_pvcs, old_osd_pvs, expected_storage_size
+):
+    """
+    Check the Ceph verification steps post resize OSD.
+    It will perform the following steps:
+    1. Check the resources state post resize OSD
+    2. Check the resources size post resize OSD
+    3. Check the Ceph state post resize OSD
+
+    Args:
+        old_osd_pods (list): The old osd pod objects before resizing the osd
+        old_osd_pvcs (list): The old osd PVC objects before resizing the osd
+        old_osd_pvs (list): The old osd PV objects before resizing the osd
+        expected_storage_size (str): The expected storage size after resizing the osd
+
+    Raises:
+        StorageSizeNotReflectedException: If the current storage size, PVCs, PVs, and ceph capacity
+            are not in the expected size
+
+    """
+    logger.info("Check the resources state post resize OSD")
+    check_resources_state_post_resize_osd(old_osd_pods, old_osd_pvcs, old_osd_pvs)
+    logger.info("Check the resources size post resize OSD")
+    check_storage_size_is_reflected(expected_storage_size)
+    logger.info("Check the Ceph state post resize OSD")
+    check_ceph_state_post_resize_osd()
+    logger.info("All the Ceph verification steps post resize osd finished successfully")
+
+
+def ceph_verification_steps_post_resize_osd(
+    old_osd_pods, old_osd_pvcs, old_osd_pvs, expected_storage_size, num_of_tries=6
+):
+    """
+    Try to execute the function 'base_ceph_verification_steps_post_resize_osd' a number of tries
```

Comment:
why are we doing that?

Reply:
Yes, right. This is regarding the comment here: https://bugzilla.redhat.com/show_bug.cgi?id=2257497#c14.

> 4. If the resize did not take effect yet, restart the OSD pods again, possibly even multiple times to get the resize to take effect. (But if it's AWS and the OSDs have already been resized previously, the resize may not be possible according to initial testing.)

Reply:
ack, thanks for the reference